secure-api-gateway-ob-uk-rcs-cloud-client/src/main/java/com/forgerock/sapi/gateway/ob/uk/rcs/cloud/client/models/ConsentClientDecisionRequest.java

```diff
@@ -49,7 +49,7 @@ public class ConsentClientDecisionRequest {
     private JWTClaimsSet jwtClaimsSet;
 
     private List<String> accountIds;
-
+    private String accountId;
```

Comment:
Why do we have an accountIds and accountId fields?

Reply:
The `ConsentClientDecisionRequest` is the shared object used to patch the consent, either the accounts consent or the payment consent, the accounts consent needs to be patch with the list of accounts chosen by the user and the payment consent needs to be patched with the account chosen by the user or with the account set as debtorAccount.
For that reason that objects, have a list of accountsId and a accountId.
Probably we can move the payment account to be set in the accountIds list as unique account and make some checks in IG to be sure that this list contains only one account before patch a payment consent.

Reply:
TBH we need to review and refactor the RCS and RS anyway. There are lots of places where we're trying to genericise something that isn't generic. Just because they all have Consent in the name, doesn't mean we should treat them generically. Same with Classes with Payment in the name. 

Happy to let this go now, on the understanding that https://github.com/SecureApiGateway/SecureApiGateway/issues/773 includes this work.

We really do need to focus on getting the blocking bug fixed here and open issues for stuff we find on the way.